slu-csci-5030/RAM_Team_SLU
# Write Unit Tests for Node.js Routes using Jest for inserting the equipment details and retreiving the details

Resolved the issue #176 

Description:
This pull request addresses the need for comprehensive unit testing of the adding and retrieving functionality in the equipmentRouter.js file. The primary goal is to ensure that equipment assets can be successfully created and retrieved from the database, with appropriate error handling for various scenarios.

Changes Made:
Added Unit Tests: Implemented unit tests using Jest and Supertest to cover the adding and retrieving functionality of the equipment router.

Tested Adding Functionality:
Valid request with all required fields.
Invalid request with missing required fields.

Successful creation of an asset.
Tested Retrieving Functionality:

Retrieving all assets.
Retrieving a specific asset by ID.

How to Test:
Clone this branch locally.
Install dependencies and set up Jest using the provided configuration.
Run the unit tests using the command: npm test.
Ensure that all tests pass successfully and verify the correctness of the test results.

Additional Notes:
The test cases cover various scenarios to ensure the robustness of the adding and retrieving functionality.
Error handling has been thoroughly tested to handle different edge cases gracefully.
Mocking of database interactions has been implemented to isolate the tests and ensure they run efficiently.
